src/server/utils/logging/attachRequestSpies.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import https from 'node:https';
+import http from 'node:http';
+import url from 'node:url';
+import onFinished from 'on-finished';
+
+import attachSpy from './attachSpy';
+
+function buildUrlObject(options, defaultProtocol) {
+  const parsedPath = url.parse(options.path || '/');
+  const protocol = options.protocol || `${defaultProtocol}:`;
+  const urlObject = {
+    auth: options.auth,
+    hostname: options.hostname || options.host || 'localhost',
+    protocol,
+    port: options.port || (protocol === 'http:' ? 80 : 443),
+    hash: parsedPath.hash,
+    path: parsedPath.path,
+    pathname: parsedPath.pathname,
+    query: parsedPath.query,
+    search: parsedPath.search,
+  };
+  if (
+    (protocol === 'http:' && urlObject.port === 80)
+    || (protocol === 'https:' && urlObject.port === 443)
+  ) {
+    delete urlObject.port;
```

Comment:
`delete` keyword has a bad performance, can we unset the `port` by assigning `null` or `undefined`???